noether_simulator/src/noether_simulator.cpp

```diff
@@ -0,0 +1,393 @@
+
+    // create look up table for point coloring
+    simulation_points->GetPointData()->SetScalars(scalars);
+
+    simulation_points_->Reset();
+    simulation_points_ = simulation_points;
+
+    displayResults();
+
+  } // end runSimulation()
+
+  void NoetherSimulator::displayResults()
+  {
+    if(!simulation_points_)
+    {
+      return;
+    }
+
+    double mean = 0;
+    double *value;
+    vtkSmartPointer<vtkDoubleArray> scalars = vtkSmartPointer<vtkDoubleArray>::New();
+    scalars =  vtkDoubleArray::SafeDownCast(simulation_points_->GetPointData()->GetScalars());
+
+    // calculate mean of data and std_dev of data
+    int number_tuples = scalars->GetNumberOfTuples();
+    for(int i = 0; i < number_tuples; ++i)
+    {
+      value = scalars->GetTuple(i);
+      mean += *value;
+    }
+    mean = mean/scalars->GetNumberOfTuples();
+    double std_dev = 0;
+    int number_scalars = scalars->GetNumberOfTuples();
+    for(int i = 0; i < number_scalars; ++i)
+    {
+      value = scalars->GetTuple(i);
+      std_dev += pow((mean - *value), 2.0);
+    }
+    std_dev = std_dev/ (scalars->GetNumberOfTuples() - 1);
+    std_dev = sqrt(std_dev);
+
+    // calculate the lower and upper limits for the display color range
+    double range[2];
+    scalars->GetRange(range);
+
+    cout << "scalar range " << range[0] << " " << range[1] << "\n";
+    double upper_bound = mean + std_dev * scalar_sigma_;
+    double lower_bound = mean - std_dev * scalar_sigma_;
+
+    // TODO: this limit assumes that the process is an additive one,
+    //       for subtractive processes, something else will need to be done
+    if(lower_bound < 0.0)
+    {
+      lower_bound = 0.0;
+    }
+
+    vtk_viewer::VTKViewer viewer;
+    std::vector<float> color2(3);
+    color2[0] = 0.1; color2[1] = 0.1; color2[2] = 0.1;
+    viewer.addPolyDataDisplay(simulation_points_, color2, lower_bound, upper_bound);
+    viewer.renderDisplay();
+  }
+
+  vtkSmartPointer<vtkMatrix4x4> NoetherSimulator::createMatrix(double pt[3], double norm[3], double derv[3])
+  {
+    // Get the point normal and derivative for creating the 3x3 transform
+    //double* norm = paths[j].line->GetPointData()->GetNormals()->GetTuple(k);
+    //double* der = paths[j].derivatives->GetPointData()->GetNormals()->GetTuple(k);
+
+    // perform cross product to get the third axis direction
+    Eigen::Vector3d u(norm[0], norm[1], norm[2]);
+    Eigen::Vector3d v(derv[0], derv[1], derv[2]);
+    Eigen::Vector3d w = u.cross(v);
+    w.normalize();
+
+    // after first cross product, u and w will be orthogonal.  Perform cross
+    // product one more time to make sure that v is perfectly orthogonal to u and w
+    v = u.cross(w);
+    v.normalize();
+
+    // TODO: the test currently uses a cylinder source, with the y-axis as the major axis
+    // for correct visualization, need to set the norm vector (u) to the y-axis
+    Eigen::Affine3d epose = Eigen::Affine3d::Identity();
```

Comment:
I'm not sure why, but I believe Affine3d is depricated for Isometry3d. I don't know the difference. 
I also forgot that you copied this code from an old version created by alex goins.